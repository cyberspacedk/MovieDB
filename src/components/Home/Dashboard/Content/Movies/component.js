/* eslint-disable react/jsx-key */
/* eslint-disable react/prop-types */
import React from 'react';
import { Layout, Row, Col, Pagination, BackTop } from 'antd';
import Card from '../../../../shared/Card';

// eslint-disable-next-line no-unused-vars
const Movies = ({ response, totalResults, history }) => (
  <Layout>
    <Layout.Content>
      <div className="top-margin">
        <Row type="flex" gutter={16} justify="center">
          <Col span={20}>
            {response &&
              response.map(item => (
                <Card
                  key={item.id}
                  id={item.id}
                  poster={item.poster_path}
                  title={item.original_title}
                  overview={item.overview}
                  history={history}
                />
              ))}
          </Col>
        </Row>

        {totalResults && (
          <Row type="flex" justify="center">
            <Col>
              <Pagination
                defaultCurrent={1}
                total={totalResults}
                className="pagination"
                defaultPageSize={20}
              />
            </Col>
          </Row>
        )}
      </div>
    </Layout.Content>
    <BackTop />
  </Layout>
);

export default Movies;
