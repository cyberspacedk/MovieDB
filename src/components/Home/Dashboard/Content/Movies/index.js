import MoviesContainer from './container';

export default MoviesContainer;
