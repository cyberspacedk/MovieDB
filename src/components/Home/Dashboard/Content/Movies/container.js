import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Movies from './component';
import {
  isError,
  isLoading,
  isEmpty,
  getSearchResponse,
  getTotalResults,
  getQuery,
  getCurrentPage,
} from '../../../../../store/search/selectors';
import { searchRequest } from '../../../../../store/search/actions';

const MoviesContainer = props => {
  const goToNextPage = page => {
    props.searchRequest(props.query, page);
  };
  return <Movies {...props} goToNextPage={goToNextPage} />;
};

MoviesContainer.propTypes = {
  query: PropTypes.string.isRequired,
  searchRequest: PropTypes.func.isRequired,
};

const mstp = state => ({
  loading: isLoading(state),
  error: isError(state),
  empty: isEmpty(state),
  movies: getSearchResponse(state),
  query: getQuery(state),
  totalResults: getTotalResults(state),
  currentPage: getCurrentPage(state),
});

const mdtp = {
  searchRequest,
};

export default compose(
  connect(
    mstp,
    mdtp,
  ),
  withRouter,
)(MoviesContainer);
