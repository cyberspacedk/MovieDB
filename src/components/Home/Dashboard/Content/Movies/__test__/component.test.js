import React from 'react';
import { shallow } from 'enzyme';
import Movies from '../component';

describe('Component: Movies', () => {
  const props = {
    loading: false,
    error: false,
    empty: false,
    movies: [{ id: 1 }],
    totalResults: 10,
    history: {
      push: jest.fn(),
    },
    goToNextPage: jest.fn(),
    currentPage: 1,
  };
  const container = shallow(<Movies {...props} />);

  it('Snapshot: should match. Field without errors', () => {
    expect(container).toMatchSnapshot();
  });
});
