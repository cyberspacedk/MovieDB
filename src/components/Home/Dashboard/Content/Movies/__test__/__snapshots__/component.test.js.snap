// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Movies Snapshot: should match. Field without errors 1`] = `
<PageLayout
  array={
    Array [
      Object {
        "id": 1,
      },
    ]
  }
  currentPage={1}
  empty={false}
  error={false}
  goToNextPage={[MockFunction]}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  loading={false}
  operations={[Function]}
  removeBox={false}
  title=""
  totalResults={10}
/>
`;
